output_firehose.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"io"
+	"log"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/firehose"
+)
+
+// FirehoseOutput plugin
+type FirehoseOutput struct {
+	fh         *firehose.Firehose
+	streamName string
+}
+
+// NewFirehoseOutput constructor for FirehoseOutput, accepts firehose stream name
+func NewFirehoseOutput(streamName string) io.Writer {
+	fatalUnlessCredentials()
+	f := new(FirehoseOutput)
+	f.fh = firehose.New(session.New(), aws.NewConfig().WithRegion(os.Getenv("AWS_REGION")))
+	f.streamName = streamName
+	return f
+}
+
+func (f *FirehoseOutput) Write(data []byte) (n int, err error) {
+	if !isOriginPayload(data) {
+		return len(data), nil
+	}
+
+	data = append(data, []byte(payloadSeparator)...)
+
+	// Gor will ignore errors here
+	f.fh.PutRecord(
+		&firehose.PutRecordInput{
+			DeliveryStreamName: aws.String(f.streamName),
+			Record: &firehose.Record{
+				Data: data,
+			},
+		},
+	)
+	return len(data), nil
+}
```

Comment:
definitely thought about, was unsure how to do it :)

Thanks for the code sample, will take a look.